### DISCUSSION TITLE
Can we do the same thing like Vedo's excellent gallery?

### DISCUSSION THREAD

**Discussion by tkoyama010** [pyvista/pyvista#1459] [Ideas]
I was really impressed the [Vedo's excellent gallery](https://vedo.embl.es/#gallery). It must be nice if we could do the same thing in PyVista project.

**Comment by akaszynski** [comment]
Yes, Vedo has great examples, and they've been rendered and displayed really well.  Our examples gallery looks plain by comparison!

**Comment by banesullivan** [comment]
It is an impressive gallery, indeed! I think PyVista and V3do are in an interesting symbiotic relationship in regards to sharing examples and yet competitive in that we're both impressed by the other and keep doing more. It makes me happy to see this one as the first in that gallery 😄 https://github.com/marcomusy/vedo/blob/master/examples/advanced/geological_model.py 

<img width="211" alt="Screen Shot 2021-12-01 at 10 16 52 PM" src="https://user-images.githubusercontent.com/22067021/144362157-3864e92d-d5a4-462f-a1e9-51f84aaf2cd0.png">


Making the examples is one of the parts of PyVista dev I enjoy the most so I may take it on to add more examples and see how I can improve our presentation of existing examples.

The tricky part is that we are sort of locked into the current structure because we have shared links to these examples in SOOOOO many places (pyvista support forum, here in discussions, issues across repos, threads on the VTK discourse, stack overflow, and more). So I want to make sure we take any approach to restructure the examples with as much caution as we would an actual API change.

With that said, there are still a lot of ways in which we can reorganize/format the examples to streamline the user experience

**Comment by tkoyama010** [comment]
I asked ChatGPT to compare PyVista and vedo. This is a very interesting analysis.

> PyVista and vedo are both Python libraries that provide tools for 3D visualization and mesh processing, but they have some differences in terms of their design philosophy, features, and use cases.
> 
> 1. Design Philosophy: PyVista focuses on providing a high-level, user-friendly interface for 3D visualization and mesh processing. It aims to provide a consistent and intuitive API for creating and manipulating 3D geometries, visualizing them with various rendering backends, and performing common mesh operations. Vedo, on the other hand, follows a more minimalist approach, providing a lightweight and flexible toolkit for creating complex visualizations with minimal boilerplate code. Vedo emphasizes on providing a concise and expressive API for creating visually appealing visualizations with less verbosity
> 
> 1. Features: Both PyVista and vedo provide a wide range of features for 3D visualization and mesh processing. PyVista offers advanced features such as volumetric rendering, cutting planes, contouring, and isosurfacing, as well as mesh operations such as slicing, mesh smoothing, and mesh decimation. PyVista also provides integration with external libraries like VTK, NumPy, and Matplotlib, allowing for more advanced customization. Vedo, on the other hand, provides a rich set of plotting functions for creating visualizations with minimal code, including options for creating surface plots, volume renderings, point clouds, streamlines, and more. Vedo also provides tools for creating animations, interactive plots, and advanced lighting effects.
> 
> 1. Use Cases: PyVista is well-suited for users who require a comprehensive and high-level interface for 3D visualization and mesh processing. It is commonly used in scientific computing, computational modeling, and data analysis tasks where complex visualizations and mesh operations are required. PyVista is also popular among the geoscience and computational fluid dynamics (CFD) communities. Vedo, on the other hand, is ideal for users who prefer a lightweight and expressive toolkit for creating visually appealing visualizations with minimal code. It is suitable for a wide range of use cases, including scientific visualization, data exploration, and interactive visualizations for presentations or interactive notebooks.
> 
> 1. Community and Documentation: Both PyVista and vedo have active communities and extensive documentation. PyVista has a larger user base and a more mature community, with comprehensive documentation, tutorials, and examples available on its website. PyVista also offers commercial support options for enterprise users. Vedo has a smaller community, but it is growing rapidly, and the library is actively maintained by its developer. Vedo also provides documentation and examples on its website, as well as a forum for community support.
> 
> In summary, PyVista and vedo are both powerful Python libraries for 3D visualization and mesh processing, but they have different design philosophies, features, and use cases. PyVista provides a high-level, comprehensive interface for 3D visualization and mesh processing, while vedo offers a lightweight and expressive toolkit for creating visually appealing visualizations with minimal code. The choice between PyVista and vedo depends on the specific needs and preferences of the user, as well as the requirements of the particular use case.
